File: Daily Exercises/Mar 2021/2021-03-20 : 150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        s=list()
        for item in tokens:
            if item=="+":
                result=s[-2]+s[-1]
            elif item=="-":
                result=s[-2]-s[-1]
            elif item=="*":
                result=s[-2]*s[-1]
            elif item=="/":
                result=int(s[-2]/s[-1])
            else:
                s.append(int(item))
                continue
            s.pop()
            s.pop()
            s.append(result)
        return s[0]

File: Daily Exercises/Mar 2021/2021-03-20 : 150. Evaluate Reverse Polish Notation/150. Evaluate Reverse Polish Notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;
        for(auto &item:tokens){
            if(isNumber(item)){
                s.push(atoi(item.c_str()));
            }else{
                int top1=s.top();
                s.pop();
                int top2=s.top();
                s.pop();
                switch(item[0]){
                    case '+':
                        s.push(top2+top1);
                        break;
                    case '-':
                        s.push(top2-top1);
                        break;
                    case '*':
                        s.push(top2*top1);
                        break;
                    case '/':
                        s.push(top2/top1);
                        break;
                }
            }
        }
        return s.top();
    }

    bool isNumber(string& token) {
        return !(token == "+" || token == "-" || token == "*" || token == "/");
    }
};
